src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.skia.cs

```diff
@@ -0,0 +1,223 @@
+
+			if (!newState.Equals(previousLayoutState))
+			{
+				if (
+					background != null ||
+					(borderThickness != Thickness.Empty && borderBrush != null)
+				)
+				{
+
+					_layerDisposable.Disposable = null;
+					_layerDisposable.Disposable = InnerCreateLayer(owner.Visual, area, background, borderThickness, borderBrush, cornerRadius);
+				}
+				else
+				{
+					_layerDisposable.Disposable = null;
+				}
+
+				_currentState = newState;
+			}
+		}
+
+		/// <summary>
+		/// Removes the added layers during a call to <see cref="UpdateLayer" />.
+		/// </summary>
+		internal void Clear()
+		{
+			_layerDisposable.Disposable = null;
+			_currentState = null;
+		}
+
+		private static IDisposable InnerCreateLayer(
+			ContainerVisual parent,
+			Rect area,
+			Brush background,
+			Thickness borderThickness,
+			Brush borderBrush,
+			CornerRadius cornerRadius)
+		{
+			var disposables = new CompositeDisposable();
+			var subVisuals = new List<Visual>();
+
+			var adjustedLineWidth = borderThickness.Top;
+			var adjustedLineWidthOffset = adjustedLineWidth / 2;
+
+			var adjustedArea = area;
+			adjustedArea.Inflate(-adjustedLineWidthOffset, -adjustedLineWidthOffset);
+
+			var compositor = parent.Compositor;
+
+			var shapeVisual = compositor.CreateShapeVisual();
+			parent.Children.InsertAtBottom(shapeVisual);
+
+			var spriteShape = compositor.CreateSpriteShape();
+
+			SkiaGeometrySource2D BuildGeometry()
+			{
+				var maxRadius = Math.Max(0, Math.Min((float)area.Width / 2 - adjustedLineWidthOffset, (float)area.Height / 2 - adjustedLineWidthOffset));
+				cornerRadius = new CornerRadius(
+					Math.Min(cornerRadius.TopLeft, maxRadius),
+					Math.Min(cornerRadius.TopRight, maxRadius),
+					Math.Min(cornerRadius.BottomRight, maxRadius),
+					Math.Min(cornerRadius.BottomLeft, maxRadius));
+
+				var geometry = new SkiaGeometrySource2D();
+
+				Brush.AssignAndObserveBrush(borderBrush, color => spriteShape.StrokeBrush = compositor.CreateColorBrush(color))
+					.DisposeWith(disposables);
+
+				geometry.Geometry.MoveTo((float)adjustedArea.GetMidX(), (float)adjustedArea.Y);
+				geometry.Geometry.ArcTo((float)adjustedArea.Right, (float)adjustedArea.Top, (float)adjustedArea.Right, (float)adjustedArea.GetMidY(), (float)cornerRadius.TopRight);
+				geometry.Geometry.ArcTo((float)adjustedArea.Right, (float)adjustedArea.Bottom, (float)adjustedArea.GetMidX(), (float)adjustedArea.Bottom, (float)cornerRadius.BottomRight);
+				geometry.Geometry.ArcTo((float)adjustedArea.Left, (float)adjustedArea.Bottom, (float)adjustedArea.Left, (float)adjustedArea.GetMidY(), (float)cornerRadius.BottomLeft);
+				geometry.Geometry.ArcTo((float)adjustedArea.Left, (float)adjustedArea.Top, (float)adjustedArea.GetMidX(), (float)adjustedArea.Top, (float)cornerRadius.TopLeft);
+
+				geometry.Geometry.Close();
+
+				if (background is LinearGradientBrush lgbBackground)
+				{
+					//var fillMask = new CAShapeLayer()
+					//{
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5872302)